packages/web-client/app/components/card-pay/issue-prepaid-card-workflow/face-value/index.ts

```diff
@@ -5,22 +5,51 @@ import { reads } from 'macro-decorators';
 import { inject as service } from '@ember/service';
 import Layer2Network from '@cardstack/web-client/services/layer2-network';
 import { fromWei, toBN, toWei } from 'web3-utils';
-import { faceValueOptions, spendToUsdRate } from '../workflow-config';
+import BN from 'web3-core/node_modules/@types/bn.js';
 import {
+  ConvertibleSymbol,
   TokenDisplayInfo,
   TokenSymbol,
 } from '@cardstack/web-client/utils/token';
-import BN from 'web3-core/node_modules/@types/bn.js';
+import { faceValueOptions, spendToUsdRate } from '../workflow-config';
 import { WorkflowCardComponentArgs } from '@cardstack/web-client/models/workflow/workflow-card';
 
+interface FaceValue {
+  spendAmount: number;
+  tokenAmount: string;
+}
+
 class FaceValueCard extends Component<WorkflowCardComponentArgs> {
   faceValueOptions = faceValueOptions;
   spendToUsdRate = spendToUsdRate;
 
   @service declare layer2Network: Layer2Network;
   @reads('args.workflowSession.state.prepaidFundingToken')
   declare fundingTokenSymbol: TokenSymbol;
-  @tracked selectedFaceValue?: number;
+  @tracked selectedFaceValue?: FaceValue;
+  @tracked options: FaceValue[] = [];
+
+  constructor(owner: unknown, args: WorkflowCardComponentArgs) {
+    super(owner, args);
+    this.getTokenAmounts('DAI', this.faceValueOptions);
+  }
+
+  async getTokenAmounts(symbol: ConvertibleSymbol, spendArr: number[]) {
+    spendArr.map(async (spendAmount) => {
+      let result = await this.layer2Network.convertFromSpend(
+        symbol,
+        spendAmount
+      );
+      let tokenAmount = parseFloat(fromWei(result)).toFixed(2);
+      this.options = [
+        ...this.options,
+        {
+          spendAmount,
+          tokenAmount,
+        },
+      ];
+    });
```

Comment:
This `map` can be a `forEach`, since we aren't returning any values.

Reply:
Another concern with this map is, there is no guaranteed resolution order of requests to the sdk, so assignment to `this.options` can happen out of order. Code below is an alternative that will always return the options in the same order as `spendArr`.

```
this.options = await Promise.all(
      spendArr.map(async (spendAmount) => {
        let result = await this.layer2Network.convertFromSpend(
          symbol,
          spendAmount
        );
        let tokenAmount = parseFloat(fromWei(result)).toFixed(2);
        return {
          tokenAmount,
          spendAmount,
        };
      })
    );
```

Reply:
Determining whether there is enough token in the assignment of the `this.options` - caveat being this does not update when the balance updates, though converting `this.options` to a getter should change that.
```
       {
          spendAmount,
          canPurchase: this.layer2Network.defaultTokenBalance?.gte(
            toBN(result)
          ),
          tokenAmount,
        },
```

Reply:
Updated my PR to be something similar to your suggestions. Decided to add the approximate symbol back because otherwise we need to display 18 digits. None of these convert to round numbers. However, the `isOptionDisabled` takes into account the real token conversion, so the option will be available to select if the user actually has enough tokens.